IFComp/lib/IFComp/Controller/Admin/Voting.pm

```diff
@@ -0,0 +1,158 @@
+package IFComp::Controller::Admin::Voting;
+use Moose;
+use namespace::autoclean;
+use Data::Dumper;
+use JSON;
+
+BEGIN { extends 'Catalyst::Controller'; }
+
+=head1 NAME
+
+IFComp::Controller::Admin::Voting
+
+=head1 DESCRIPTION
+
+A controller for voting reports.
+
+=head1 METHODS
+
+=cut
+
+
+=head2 index
+
+=cut
+
+sub index :Path :Args(0) {
```

Comment:
Please make this and all other /admin/foo endpoints Chained actions, which chain off the root admin action. (See IFComp::Controller::Entry for examples of chained actions.)

Doing this will let you apply role-based authorization to to root admin action and then have it automatically applied to all these actions as well.